pkg/azschedulerextender/main.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"encoding/json"
+	"math/rand"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+	schedulerapi "k8s.io/kube-scheduler/extender/v1"
+	"time"
+)
+
+func init() {
+	klog.InitFlags(nil)
+}
+
+var (
+	metricsAddress = flag.String("metrics-address", "0.0.0.0:29604", "export the metrics")
+)
+
+const (
+	filterRequestStr     = "filter"
+	prioritizeRequestStr = "prioritize"
+	pingRequestStr = "ping"
+)
+
+func main() {
+	flag.Parse();
+	exportMetrics()
+	klog.V(5).Infof("Starting azschedulerextender ...")
+	server := &http.Server{Addr: ":8080"}
+	http.HandleFunc("/", serveSchedulerTraffic)
+	if err := server.ListenAndServe(); err != http.ErrServerClosed {
+		klog.Fatalf("Exiting. Error starting extender server: %v", err)
+	}
+	klog.V(5).Infof("Exiting azschedulerextender ...")
+	os.Exit(0)
+}
+
+func serveSchedulerTraffic(response http.ResponseWriter, request *http.Request) {
```

Comment:
Consider `handleSchedulerRequest` for the name here, but really I would recommend is registering the separate handler functions for each verb instead of using the substring matching below. The current matching is too loose and would accept paths like `/filtering/something` or `prioritizing/nothing` as valid requests.

Reply:
updated.

Reply:
The behaviour is described in the [ServeMux documentation](https://pkg.go.dev/net/http#ServeMux). My interpretation is that it is not a prefix match.

Reply:
Thanks for sharing the documentation. I was getting confused with the behavior for the rooted paths /.../, in our case we are not registering any such path, so may be your interpretation is correct\better than mine. :)